src/pickley/__init__.py

```diff
@@ -665,6 +666,16 @@ def program_version(path, logger=None):
             if r.succeeded:
                 return PickleyConfig.parsed_version(r.output or r.error)
 
+    @staticmethod
+    def installed_sibling_folders(canonical_name):
+        """Sibling installations of the form '<canonical_name>-<version>'."""
+        regex = re.compile(r"^(.+)-(\d+[.\d+]+)$")
```

Comment:
This doesn't handle any pre or post releases, is there a reason for it to be overly strict?

Is there a reason to allow `+` in the character class, are you trying to support local identifiers?  If you intend to allow them, they are not restricted to digits.

[pypi-simple uses](https://github.com/jwodder/pypi-simple/blob/a357d83ad83ac54a81c0abb6097b9858e424f222/src/pypi_simple/filenames.py#L9) uses `VERSION_NODASH` which is short and overly permissive.

Reply:
Ouch! Good point, this old code needs to be adapted! As-is, it would never clean up a `1.0rc2` for example...